[DX-768] Tyk Charts release notes : Reviewing For Release On GHub (Do not merge) (#3413)
TykTechnologies/tyk-docs: tyk-docs/content/developer-support/tyk-charts-release-notes.md

- `tyk-single-dc` chart is renamed to `tyk-stack`. This change is part of the terminology alignment initiatives where Tyk is standardising how we refer to the components. If you have previously used `tyk-single-dc`, please delete the release and reinstall using `tyk-stack`. Please refer to change log below for enhancements and fixes that are added to the new chart.
- Renamed parameter `backend` to `storageType` in `tyk-dashboard`
#### Deprecation
- `tyk-mdcb-data-plane` is now marked as deprecated.
- `tyk-enterprise-portal` is now marked as deprecated.
- `tyk-single-dc` is now marked as deprecated.

#### Upgrade instructions

For renamed charts, please delete the release and reinstall using the new chart.
```
helm delete [RELEASE_NAME] tyk-helm/tyk-mdcb-data-plane
helm install [RELEASE_NAME] tyk-helm/tyk-data-plane
```

For other charts, you can use `helm upgrade` to upgrade your release.
```
helm upgrade [RELEASE_NAME] tyk-helm/tyk-oss
```

#### Release Highlights
This version upgrades Tyk Gateway and Tyk Dashboard to v5.2.0, Tyk Pump to 1.8.3 and Tyk Portal to 1.7.0.

> ##### Security enhancements

Author: Suggestion committed

> This version introduces a few security enhancements. It adds configuration options to configure SSL in dashboard, and support of `insecureSkipVerify` option for all charts to bypass verification for self-signed certificates. For security best practices, we now support use of secret to pass sensitive fields including admin credentials, license keys, database connection string, and remote control plane connection details via secrets.

Author: Updated

##### New Gateway parameters

> This version enhances Gateway charts by introducing more parameters, like `containerPort`, `enalyticsEnabled`, `analyticsConfigType`, `hashkeyFunction` for Gateway. 

Author: Updated

##### New Portal parameters
The latest tyk-dev-portal beta chart has full support of all storage type options: `fs`, `db`, and `s3`.

#### Compatibility Notes
This release is tested on Kubernetes 1.26.3, 1.25.2, 1.24.6, 1.23.12, 1.22.15, 1.21.14, 1.20.15, Tyk Gateway v5.2.0, Tyk Dashboard v5.2.0, Tyk Pump v1.8.3, and Tyk Portal v1.7.0.

[tyk-pump-1.1.0](#tyk-pump-1.1.0) | [tyk-gateway-1.1.0](#tyk-gateway-1.1.0) | [tyk-oss-1.1.0](#tyk-oss-1.1.0) | [tyk-data-plane-1.0.0](#tyk-data-plane-1.0.0) | [tyk-dashboard-1.0.0-beta6](#tyk-dashboard-1.0.0-beta6) | [tyk-bootstrap-1.0.0-beta6](#tyk-bootstrap-1.0.0-beta6) | [tyk-dev-portal-1.0.0-beta1](#tyk-dev-portal-1.0.0-beta1) | [tyk-stack-1.0.0-beta1](#tyk-stack-1.0.0-beta1)

---
## tyk-pump-1.1.0
#### Changelog
##### Added

> - Added parameter `.global.mongo.driver` to configure which Mongo Driver to use.

Author: Updated

- Added new options to `pump.backend` parameter. Users can enable specific Mongo & Postgres pumps: `mongo-aggregate`, `mongo-selective`, `postgres-aggregate`, `postgres-pump`
- Added `global.remoteControlPlane.useSecretName` parameter to allows user to pass control plane connection details via Kubernetes secrets

> - Added support for `containerSecurityContext` configuration. This is required as k8s and openshift versions require some of these values to be set.

Author: Updated

- Added support for `imagePullSecret` so user can pull an image from a private container image registry or repository
##### Changed
- Updated Pump default image tag to v1.8.3
##### Fixed

> - Fixed use the resources specified in values.yaml

Author: Updated

## tyk-gateway-1.1.0
#### Changelog

> ##### Added

Author: Updated

- Added `Horizontal Pod Autoscaler` specs for gateway deployments, allowing users to easily enable automatic scaling by CPU utilisation, memory utilisation, or custom metrics.

> - Added `insecureSkipVerify` option for gateway under `gateway.tls` section.

Reviewer: Add further explanation of what this does and why the user might need it

Author: Updated

- Added `global.remoteControlPlane.useSecretName` parameter to allows user to pass control plane connection details via Kubernetes secrets

> - Added support for `containerSecurityContext` configuration. This is required as k8s and openshift versions require some of these values to be set.

Author: Updated

> - Added support for `containerSecurityContext` configuration. This is required as k8s and openshift versions require some of these values to be set.

Reviewer: Should we also add the OpenShift link? https://access.redhat.com/documentation/en-us/openshift_container_platform/4.12/html/authentication_and_authorization/managing-pod-security-policies

- Added `containerPort` parameter for gateway to allow for different values to be set for port and targetPort
- Added support for `imagePullSecret` so user can pull an image from a private container image registry or repository
- Added parameter `analyticsEnabled` to enable or disable analytics in Gateway. It is set to "" by default which means it will be enabled or disabled based on pump installations.
- Added `hashKeyFunction` parameter for Gateway. Default to `murmur128`
##### Changed
- Updated Gateway default image tag to v5.2.0

> - Removed setting of obsolete environment variable TYK_GW_OPTIMISATIONSUSEASYNCSESSIONWRITE

Reviewer: Is this deprecated and will it be a breaking change?

Author: This environment variable is not used by Tyk anymore so we do not need to set it.

Reviewer: Ok thanks Carol, anything such as that we can mention in deprecated changes section 

https://tyktech.atlassian.net/wiki/spaces/EN/pages/2131918959/Guidelines+For+Writing+Tyk+Release+Notes#Deprecations

## tyk-oss-1.1.0
#### Changelog
##### Added
- Added new options to `pump.backend` parameter. Users can enable specific Mongo & Postgres pumps: `mongo-aggregate`, `mongo-selective`, `postgres-aggregate`, `postgres-pump`

> - Added `Horizontal Pod Autoscaler` specs for gateway deployments, allowing users to easily enable automatic scaling by CPU utilisation, memory utilisation, or custom metrics.

Reviewer: ```suggestion
- Added `Horizontal Pod Autoscaler` specs for gateway deployments, allowing users to easily enable automatic scaling by CPU utilisation, memory utilisation or custom metrics.
```

Author: Updated

- Added `insecureSkipVerify` option for gateway under `gateway.tls` section.

> - Added support for `containerSecurityContext` configuration. This is required as k8s and openshift versions require some of these values to be set.- Added `containerPort` parameter for gateway to allow for different values to be set for port and targetPort

Reviewer: ```suggestion
- Added support for `containerSecurityContext` configuration. This is required as K8s and OpenShift versions require some of these values to be set.- Added `containerPort` parameter for gateway to allow for different values to be set for port and targetPort
```

Author: Suggestion committed

#### Breaking Changes
- `tyk-mdcb-data-plane` chart is renamed to `tyk-data-plane`. This change is part of the terminology alignment initiatives where Tyk is standardising how we refer to the components. If you have previously used `tyk-mdcb-data-plane`, please delete the release and reinstall using `tyk-data-plane`. Please refer to change log below for enhancements and fixes that are added to the new chart.
#### Deprecation
- `tyk-mdcb-data-plane` is now marked as deprecated.
#### Changelog
##### Added
- Added new options to `pump.backend` parameter. Users can enable specific Mongo & Postgres pumps: `mongo-aggregate`, `mongo-selective`, `postgres-aggregate`, `postgres-pump`
- Added `Horizontal Pod Autoscaler` specs for gateway deployments, allowing users to easily enable automatic scaling by CPU utilisation, memory utilisation, or custom metrics.
- Added `insecureSkipVerify` option for gateway under `gateway.tls` section.
- Added `global.remoteControlPlane.useSecretName` parameter to allows user to pass control plane connection details via Kubernetes secrets. For detail uses of secrets, see [README](https://github.com/TykTechnologies/tyk-charts/blob/main/tyk-data-plane/README.md)

> - Added support for `containerSecurityContext` configuration. This is required as k8s and openshift versions require some of these values to be set.

Reviewer: ```suggestion
- Added support for `containerSecurityContext` configuration. This is required as K8s and OpenShift versions require some of these values to be set.
```

Reviewer 2: good place to add an external link

Author: Suggestion committed

Author: Updated

##### Changed
- Changed default value of `analyticsConfigType` to empty, so that gateway will send analytics to Pump by default.
- Updated Gateway default image tag to v5.2.0
- Updated Pump default image tag to v1.8.3

## tyk-dashboard-1.0.0-beta6
#### Breaking Changes
- Renamed parameter `backend` to `storageType`
#### Changelog
##### Added
- Added parameter `.global.mongo.driver` to configure which Mongo Driver to use.
- Added support for SSL configurations in `dashboard.tls` section. User can reference the TLS certificate as secret.
- Added `insecureSkipVerify` option for dashboard under `dashboard.tls` section.
- Added `global.adminUser.useSecretName` parameter to allows user to pass Admin Credentials via Kubernetes secrets
- Added support for `containerSecurityContext` configuration. This is required as k8s and openshift versions require some of these values to be set.
- Added support for `imagePullSecret` so user can pull an image from a private container image registry or repository
##### Changed
- Changed default storageType to `postgres`
- Removed `components/tyk-dashboard/certs/` folder as it is not required.
- Updated Dashboard default image tag to v5.2.0
##### Fixed

> - Fixed duplicate setting of `TYK_DB_ENABLEAGGREGATELOOKUPS` environment variabe.

Author: Fixed

> - Fixed value of TYK_DB_TYKAPI_HOST environment variable

Reviewer: Add explanation of the issue, i.e. in what way was value faulty? What does the behaviour of the fix now enable for the user?

Author: Updated

> - Fixed gateway port name in dashboard deployment

Reviewer: Add explanation of the issue, in what way was the port name faulty? What does the behaviour of the fix now enable for the user?

Author: Updated

Reviewer: Hi Carol, has this item now been removed from the change-log?

- Fixed gateway port name in dashboard deployment

## tyk-bootstrap-1.0.0-beta6
#### Changelog
##### Added
- Added a pre-hook to validate Dashboard license before bootstrapping starts

> - Added support for `insecureSkipVerify` option for the `tyk-k8s-bootstrap`

Author: This line is removed

- Added parameter `sslInsecureSkipVerify`. When set to `true`, it will skip validating the SSL certificates. Usually needed when using self-signed certs.

> - Make tyk-bootstrap image repository information configurable

Author: Updated

- Added `global.adminUser.useSecretName` parameter to allows user to pass Admin Credentials via Kubernetes secrets
- Support retrieving Dashboard License from secret `global.secrets.useSecretName` to protect confidential fields with Kubernetes secrets

## tyk-dev-portal-1.0.0-beta1
#### Breaking Changes
- `tyk-enterprise-portal` chart is renamed to `tyk-dev-portal`. This change is part of the terminology alignment initiatives where Tyk is standardising how we refer to the components. If you have previously used `tyk-enterprise-portal`, please delete the release and reinstall using `tyk-dev-portal`. Please refer to change log below for enhancements and fixes that are added to the new chart.
#### Deprecated
- `tyk-enterprise-portal` is now marked as deprecated.
#### Changelog
##### Added

> - Added parameter `kind` for portal. User can choose to deploy Tyk developer portal as StatefulSet or Deployment. Default as StatefulSet.

Reviewer: ```suggestion
- Added parameter `kind` for portal. User can choose to deploy Tyk Developer Portal as StatefulSet or Deployment. Default as StatefulSet.
```

Capitalise remaining instances below for Tyk Developer Portal

Author: Updated

- Added support for `fs` portal storage type. User can mount existing PVC to Tyk developer portal.

> - Added support for `extraVolume` and `extraVolumeMount`.

Reviewer: How does this benefit users? What is the use case?

Author: Updated

- Added `global.adminUser.useSecretName` parameter to allows user to pass Admin Credentials via Kubernetes secrets

> - Added support for setting Portal licence and storage connection string via `global.secrets.useSecretName`

Reviewer: Is it possible to set both the license and storage connection string with this config value?

Author: yes. User just need to set one secret. Added link to docs on how to configure the secret for different credential or secret values.

> - Added support for `containerSecurityContext` configuration. This is required as k8s and openshift versions require some of these values to be set.

Reviewer: ```suggestion
- Added support for `containerSecurityContext` configuration. This is required as K8s and OpenShift versions require some of these values to be set.
```

Author: Updated

- Added support for `imagePullSecret` so user can pull an image from a private container image registry or repository

> - Added support for Portal storage type `db` in `storage.type`.

Reviewer: Can we link to docs here? What are the existing storage type settings other than database? What type of information is stored in the database?

Author: Updated

##### Changed
##### Fixed
- Updated Portal default image tag to v1.7.0

> - Fixed environment variable setting for s3 storage

Reviewer: What environment variable was faulty and what issue was this causing for users? How does the fix now help users?

Author: Updated

## tyk-stack-1.0.0-beta1
#### Breaking Changes
- `tyk-single-dc` chart is renamed to `tyk-stack`. This change is part of the terminology alignment initiatives where Tyk is standardising how we refer to the components. If you have previously used `tyk-single-dc`, please delete the release and reinstall using `tyk-stack`. Please refer to change log below for enhancements and fixes that are added to the new chart.
- Renamed parameter `backend` to `storageType`
#### Deprecated
- `tyk-single-dc` is now marked as deprecated.

#### Changelog
##### Added

> - Added parameter `.global.mongo.driver` to configure which Mongo Driver to use.

Reviewer: Maybe link to docs for supported driver settings?

Author: Updated

- Added new options to `pump.backend` parameter. Users can enable specific Mongo & Postgres pumps: `mongo-aggregate`, `mongo-selective`, `postgres-aggregate`, `postgres-pump`
- Added `Horizontal Pod Autoscaler` specs for gateway deployments, allowing users to easily enable automatic scaling by CPU utilisation, memory utilisation, or custom metrics.

> - Added parameter `kind` for portal. User can choose to deploy Tyk developer portal as StatefulSet or Deployment. Default as StatefulSet.

Reviewer: Can we link to docs here if available? If not add a brief explanation of what StatefulSet and Deployment do.

Author: Updated

- Added support for `fs` portal storage type. User can mount existing PVC to Tyk developer portal.
- Added `insecureSkipVerify` option for gateway under `gateway.tls` section.
- Added `insecureSkipVerify` option for dashboard under `dashboard.tls` section.
- Added support for SSL configurations in `dashboard.tls` section. User can reference the TLS certificate as secret.
- Make tyk-bootstrap image repository information configurable
- Added `global.adminUser.useSecretName` parameter to allows user to pass Admin Credentials via Kubernetes secrets. For detail uses of secrets, see [README](https://github.com/TykTechnologies/tyk-charts/blob/main/tyk-stack/README.md)
- Added support for setting Portal licence and storage connection string via `global.secrets.useSecretName`. For detail uses of secrets, see [README](https://github.com/TykTechnologies/tyk-charts/blob/main/tyk-stack/README.md)

> - Added `global.remoteControlPlane.useSecretName` parameter to allows user to pass control plane connection details via Kubernetes secrets. For detail uses of secrets, see [README](https://github.com/TykTechnologies/tyk-charts/blob/main/tyk-stack/README.md)

Reviewer: ```suggestion
- Added `global.remoteControlPlane.useSecretName` parameter to allow users to pass control plane connection details via Kubernetes secrets. For further details regarding uses of secrets, see [README](https://github.com/TykTechnologies/tyk-charts/blob/main/tyk-stack/README.md)
```

Add README link to other change-log items above that use `global.secrets.useSecretName`.

> - Added `global.remoteControlPlane.useSecretName` parameter to allows user to pass control plane connection details via Kubernetes secrets. For detail uses of secrets, see [README](https://github.com/TykTechnologies/tyk-charts/blob/main/tyk-stack/README.md)

Reviewer: ```suggestion
- Added `global.remoteControlPlane.useSecretName` parameter to allow users to pass control plane connection details via Kubernetes secrets. For detail uses of secrets, see [README](https://github.com/TykTechnologies/tyk-charts/blob/main/tyk-stack/README.md)
```

Author: Suggestion committed

> - Added support for `containerSecurityContext` configuration. This is required as k8s and openshift versions require some of these values to be set.

Reviewer: ```suggestion
- Added support for `containerSecurityContext` configuration. This is required as K8s and OpenShift versions require some of these values to be set.
```

Author: Updated

- Added `containerPort` parameter for gateway to allow for different values to be set for port and targetPort
- Added support for `imagePullSecret` so user can pull an image from a private container image registry or repository

> - Added support for Portal storage type `db` in `storage.type`.

Reviewer: Same as previously commented, what are the other values for `storage.type` other than database storage? What information is stored in the database? Are there any docs that we can link to?

Author: Updated

- Added parameter `analyticsEnabled` to enable or disable analytics in Gateway. It is set to "" by default which means it will be enabled or disabled based on pump installations.
##### Changed

> - Changed default storageType to `postgres`

Reviewer: What was previous default storage type and how does this change benefit users? Did the majority of users request the change?

Author: It was mongo previously. The commercial team has decided to recommend Postgres as default over MongoDB.

Author: Updated

> - Changed default pump backend to `prometheus` and `global.StorageType`

Reviewer: What was previous default storage type and how does this change benefit users? Did the majority of users request the change?

Author: Updated

- Updated Gateway default image tag to v5.2.0
- Updated Pump default image tag to v1.8.3
- Updated Dashboard default image tag to v5.2.0
- Updated Portal default image tag to v1.7.0
##### Fixed

> - Fixed default value of `storage.s3.endpoint` and `storage.s3.bucket`

Reviewer: What was the previous default values for `storage.s3.endpoint` and `storage.s3.bucket` that was causing an issue for users? What are the new default values and how does this help users?

Author: This line is not required now.

---
## tyk-pump-1.1.0
#### Changelog
##### Added
- Added parameter `.global.mongo.driver` to configure which [Mongo Driver](https://tyk.io/docs/tyk-dashboard/configuration/#mongo_driver) to use.
- Added new options to `pump.backend` parameter. Users can enable specific Mongo & Postgres Pumps: `mongo-aggregate`, `mongo-selective`, `postgres-aggregate`, `postgres-pump`.
- Added `global.remoteControlPlane.useSecretName` parameter to allows user to pass control plane connection details via Kubernetes secrets.
- Added support for `containerSecurityContext` configuration. This is required as *K8s* and *OpenShift* versions require the [security context for container](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) to be set.
- Added support for `imagePullSecret` so user can pull an image from a private container image registry or repository.
##### Changed
- Updated Pump default image tag to v1.8.3.
##### Fixed
- Fixed typo in Pump deployment template to pick up the correct field `pump.resources` specified in values.yaml.

## tyk-gateway-1.1.0
#### Changelog
##### Added

> - Added `Horizontal Pod Autoscaler` specs for gateway deployments, allowing users to easily enable automatic scaling by CPU utilisation, memory utilisation or custom metrics.

Reviewer: ```suggestion
- Added `Horizontal Pod Autoscaler` specs for Gateway deployments, allowing users to easily enable automatic scaling by CPU utilisation, memory utilisation or custom metrics.
```

- Added support for `containerSecurityContext` configuration. This is required as *K8s* and *OpenShift* versions require the [security context for container](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) to be set.
- Added `containerPort` parameter for Gateway to allow for different values to be set for port and targetPort.
- Added support for `imagePullSecret` so user can pull an image from a private container image registry or repository.
- Added parameter `analyticsEnabled` to enable or disable analytics in Gateway. It is set to "" by default which means it will be enabled or disabled based on Pump installations.
- Added `hashKeyFunction` parameter for Gateway. Default to `murmur128`.
##### Changed
- Updated Gateway default image tag to v5.2.0.
- Removed setting of obsolete environment variable TYK_GW_OPTIMISATIONSUSEASYNCSESSIONWRITE.

## tyk-oss-1.1.0
#### Changelog
##### Added
- Added new options to `pump.backend` parameter. Users can enable specific Mongo & Postgres Pumps: `mongo-aggregate`, `mongo-selective`, `postgres-aggregate`, `postgres-pump`.

> - Added `Horizontal Pod Autoscaler` specs for gateway deployments, allowing users to easily enable automatic scaling by CPU utilisation, memory utilisation or custom metrics.

Reviewer: ```suggestion
- Added `Horizontal Pod Autoscaler` specs for Gateway deployments, allowing users to easily enable automatic scaling by CPU utilisation, memory utilisation or custom metrics.
```

- Added support for `containerSecurityContext` configuration. This is required as *K8s* and *OpenShift* versions require the [security context for container](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) to be set.
- Added `containerPort` parameter for Gateway to allow for different values to be set for port and targetPort.
- Added support for `imagePullSecret` so user can pull an image from a private container image registry or repository.
- Added parameter `analyticsEnabled` to enable or disable analytics in Gateway. It is set to "" by default which means it will be enabled or disabled based on Pump installations.
##### Changed
- Changed default value of `analyticsConfigType` to empty, so that Gateway will send analytics to Pump by default.
- Updated Gateway default image tag to v5.2.0.
- Updated Pump default image tag to v1.8.3.

## tyk-dashboard-1.0.0-beta6
#### Breaking Changes
- Renamed parameter `backend` to `storageType`.
#### Changelog
##### Added

> - Added parameter `.global.mongo.driver` to configure which Mongo Driver to use.

Reviewer: ```suggestion
- Added parameter `.global.mongo.driver` to configure which [Mongo Driver]({{< ref "/tyk-dashboard/configuration#mongo_driver" >}}) to use.
```

> - Added parameter `.global.mongo.driver` to configure which [Mongo Driver]({{< ref "/tyk-dashboard/configuration#mongo_driver" >}}) to use.

Reviewer: ```suggestion
- Added parameter `.global.mongo.driver` to configure which [Mongo Driver](https://tyk.io/docs/tyk-dashboard/configuration/#mongo_driver) to use.
```

  analyticsEnabled: ""
```

```
  # used to decide whether to send the results back directly to Tyk without a hybrid pump
  # if you want to send analytics to control plane instead of pump, change analyticsConfigType to "rpc"
  analyticsConfigType: ""
```

```
  # hashKeyFunction property is used to specify the Key hashing algorithm.
  # Possible values: murmur64, murmur128, sha256.
  hashKeyFunction: murmur128
```

##### New Portal parameters
The latest tyk-dev-portal beta chart has full support of all storage type options: `fs`, `db`, and `s3`.

#### Compatibility Notes
This release is tested on Kubernetes 1.26.3, 1.25.2, 1.24.6, 1.23.12, 1.22.15, 1.21.14, 1.20.15, Tyk Gateway v5.2.1, Tyk Dashboard v5.2.1, Tyk Pump v1.8.3, and Tyk Portal v1.7.0.

[tyk-pump-1.1.0](#tyk-pump-1.1.0) | [tyk-gateway-1.1.0](#tyk-gateway-1.1.0) | [tyk-oss-1.1.0](#tyk-oss-1.1.0) | [tyk-data-plane-1.0.0](#tyk-data-plane-1.0.0) | [tyk-dashboard-1.0.0-beta6](#tyk-dashboard-1.0.0-beta6) | [tyk-bootstrap-1.0.0-beta6](#tyk-bootstrap-1.0.0-beta6) | [tyk-dev-portal-1.0.0-beta1](#tyk-dev-portal-1.0.0-beta1) | [tyk-stack-1.0.0-beta1](#tyk-stack-1.0.0-beta1)

---

> ## tyk-pump-1.1.0

Reviewer 2: @caroltyk @dcs3spp silly question - why do we have the titles with hyphens? in the GW we have a properly capitalised title https://github.com/TykTechnologies/tyk/releases/tag/v5.2.1

Reviewer: @letzya @caroltyk I think these are the chart names, yes?

Reviewer 2: In the release page in tyk-charts, this is the title `Tyk Chart v1.0.0`
It looks clearer as follows
## Tyk Pump v1.1.0

Author: These are chart names.

For helm chart, I see the usual customs is to have the chart name followed by versions. The top helm chart projects (https://github.com/topics/helm-charts) follow this patterns: "kube-prometheus-stack-52.1.0", "argo-rollouts-2.32.1", "airflow-8.8.0"

Non-helm chart projects are more varied (https://github.com/topics/cncf): "Kubernetes v1.25.15" "v1.8.0" "Fluentd 1.1.0"
